packages/web/src/exporters/otlp.ts

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2024 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { diag } from '@opentelemetry/api';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import { NOOP_ATTRIBUTES_TRANSFORMER, NATIVE_XHR_SENDER, NATIVE_BEACON_SENDER, SplunkExporterConfig } from './common';
+import { IExportTraceServiceRequest } from '@opentelemetry/otlp-transformer';
+import { ReadableSpan } from '@opentelemetry/sdk-trace-base';
+
+export class SplunkOTLPTraceExporter extends OTLPTraceExporter {
+  protected readonly _onAttributesSerializing: SplunkExporterConfig['onAttributesSerializing'];
+  protected readonly _xhrSender: SplunkExporterConfig['xhrSender'] = NATIVE_XHR_SENDER;
+  protected readonly _beaconSender: SplunkExporterConfig['beaconSender'] = typeof navigator !== 'undefined' && navigator.sendBeacon ? NATIVE_BEACON_SENDER : undefined;
+
+  constructor(options: SplunkExporterConfig) {
+    super(options);
+    this._onAttributesSerializing = options.onAttributesSerializing || NOOP_ATTRIBUTES_TRANSFORMER;
+  }
+
+  convert(spans: ReadableSpan[]): IExportTraceServiceRequest {
+    // Changes: Add attribute serializing hook to remove data before export
+    spans = spans.map(span => {
+      // @ts-expect-error Yep we're overwriting a readonly property here. Deal with it
+      span.attributes = this._onAttributesSerializing ? this._onAttributesSerializing(span.attributes, span) : span.attributes;
```

Comment:
this._onAttributesSerializing alway true no?

Reply:
should be, tho types says maybe